### DISCUSSION TITLE
Why does the pc terminal run well but not on the mobile phone?

### DISCUSSION THREAD

**Discussion by callmeitachi** [leptos-rs/leptos#2538] [Q&A]
i have the following code:
use leptos::*;
use leptos_router::use_navigate;
use serde_json::json;
use gloo_net::http::Request;
use gloo_storage::{LocalStorage, Storage};
use gloo_storage::SessionStorage;
use leptos::html::{Form,A};
#[component]
pub fn Login() -> impl IntoView {
    let navigation=use_navigate();
    let (username,set_username)=create_signal("".to_string());
    let (password,set_password)=create_signal("".to_string());
    let on_click=move |ev:ev::MouseEvent| {
        ev.prevent_default();
        let navigation=navigation.clone();
        spawn_local(async move {
            let username=username.get();
        let password=password.get();
        let json_data=json!({
            "username":username,
            "password":password,
        });
        let form_data=json_data.to_string();
        let res=Request::post("http://127.0.0.1:5000/login1")
                        .header("Content-Type","application/json")
                        .body(form_data)
                        .unwrap()
                        .send()
                        .await
                        .unwrap();
        if res.status() == 200 {
            let token=res.text().await.unwrap();
            SessionStorage::set("Authorization", token).unwrap();
            navigation("/home",Default::default());
        } else {
            navigation("/login",Default::default());
        }

        });
    };
    view! {
        <div class="container">
            
            <div class="form">
                <h1>"system"</h1>
                <input type="text" name="username" on:input=move |ev| {
                    set_username.set(event_target_value(&ev));
                    logging::log!("{}",username.get());
                } prop:value=username placeholder="username" />
                <input type="password" name="password" placeholder="password" on:input=move |ev| {
                    set_password.set(event_target_value(&ev));
                    logging::log!("{}",password.get());
                } />
                <button on:click=on_click >"login"</button>
            </div>
        </div>
    }
}

**Comment by gbj** [comment]
I don't understand what you're asking, sorry.

**Reply by callmeitachi** [reply]
why it doesn't work in mobile phone when i click the button?what should i do?

**Comment by mfenniak** [comment]
It's a little hard to answer a question like this since you haven't described what you are expecting, what you are actually experiencing, and what you've tried to do to resolve the issue.

However, I can make a guess.  Your application is trying to access "127.0.0.1:5000", which is a network port 5000 on the **local device**.  I would imagine that it works on your computer because you have something running that is listening and operating on port 5000, and responding to the "/login1" HTTP request.  However, that likely isn't the case on your phone.  The way you created this, it will always access "this device".

**Reply by callmeitachi** [reply]
Sorry，I'm not good at English.And your describe is right,so what resolve it？
